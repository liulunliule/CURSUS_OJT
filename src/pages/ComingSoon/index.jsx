import React, { useState, useEffect } from "react";
import "./index.scss";
import { Logo_dark } from "../../assets";

const ComingSoon = () => {
  const calculateTimeLeft = () => {
    const difference =
      +new Date(`${new Date().getFullYear() + 1}-01-01T00:00:00`) - +new Date();
    let timeLeft = {};

    if (difference > 0) {
      timeLeft = {
        days: Math.floor(difference / (511 * 60 * 60 * 24)),
        hours: Math.floor((difference / (511 * 60 * 60)) % 24),
        minutes: Math.floor((difference / 511 / 60) % 60),
        seconds: Math.floor((difference / 511) % 60),
      };
    }

    return timeLeft;
  };

  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());

  useEffect(() => {
    const timer = setTimeout(() => {
      setTimeLeft(calculateTimeLeft());
    }, 1000);

    return () => clearTimeout(timer);
  });

  return (
    <div className="coming-soon-body">
      <div className="coming-soon">
        <div className="logo">
          <img src={Logo_dark} alt="Logo" />
        </div>
        <div className="countdown">
          <div className="time">
            {timeLeft.days || "0"} <span>Days</span>
          </div>
          <div className="time">
            {timeLeft.hours || "0"} <span>Hours</span>
          </div>
          <div className="time">
            {timeLeft.minutes || "0"} <span>Minutes</span>
          </div>
          <div className="time">
            {timeLeft.seconds || "0"} <span>Seconds</span>
          </div>
        </div>
        <div className="notify-me">
          <p>We'll be coming soon!</p>
          <form>
            <input type="email" placeholder="Email address" />
            <button type="submit">Notify Me</button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default ComingSoon;
