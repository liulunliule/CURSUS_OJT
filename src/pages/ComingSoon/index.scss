.coming-soon-body {
  margin: 0;
  padding: 0;
  font-family: Arial, sans-serif;
  background-color: #1a1a1a;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  overflow: hidden;
  transform: scale(1);
  transform-origin: center;
}

.coming-soon {
  width: 100%;
  max-width: 1920px;
  max-height: 1080px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;

  .logo {
    margin-bottom: 40px;
  }

  .countdown {
    display: flex;
    justify-content: center;
    gap: 50px;
    font-size: 6em;
    margin-bottom: 40px;

    .time {
      display: flex;
      flex-direction: column;
      align-items: center;
      font-size: 1em;
      font-weight: bold;

      span {
        font-size: 0.25em;
        margin-top: 10px;
        color: #999;
      }
    }
  }

  .notify-me {
    p {
      margin-top: 30px;
      font-size: 1.5em;
      margin-bottom: 20px;
    }

    form {
      display: flex;
      justify-content: center;
      margin-bottom: 40px;
      height: 30px;
      font-size: 1em;

      input {
        padding: 20px;
        width: 300px;
        border: none;
        border-radius: 10px;
        font-size: 1em;
        margin-right: 20px;
        background: white;
        transition: background 0.3s;

        &:focus {
          outline: none;
          font-weight: bold;
        }
      }

      button {
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 20px;
        font-size: 1em;
        background-color: red;
        border: none;
        border-radius: 10px;
        color: #fff;
        cursor: pointer;
        transition: background-color 0.3s;
      
        &:hover {
          background-color: black;
        }
      }
    }
  }
}