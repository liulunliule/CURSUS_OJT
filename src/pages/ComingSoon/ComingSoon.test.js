import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ComingSoon from '.';

test('renders Coming Soon component correctly', () => {
  const { asFragment } = render(<ComingSoon />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders countdown timer correctly', () => {
  render(<ComingSoon />);
  
  expect(screen.getByText(/Days/i)).toBeInTheDocument();
  expect(screen.getByText(/Hours/i)).toBeInTheDocument();
  expect(screen.getByText(/Minutes/i)).toBeInTheDocument();
  expect(screen.getByText(/Seconds/i)).toBeInTheDocument();
});

test('renders Notify Me section correctly', () => {
  render(<ComingSoon />);
  
  expect(screen.getByText("We'll be coming soon!")).toBeInTheDocument();
  expect(screen.getByPlaceholderText('Email address')).toBeInTheDocument();
  expect(screen.getByText('Notify Me')).toBeInTheDocument();
});

test('email input has correct placeholder text', () => {
  render(<ComingSoon />);
  
  const emailInput = screen.getByPlaceholderText('Email address');
  expect(emailInput).toBeInTheDocument();
  expect(emailInput).toHaveAttribute('type', 'email');
});