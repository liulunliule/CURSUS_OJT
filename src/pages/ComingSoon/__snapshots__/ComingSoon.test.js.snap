// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Coming Soon component correctly 1`] = `
<DocumentFragment>
  <div
    class="coming-soon-body"
  >
    <div
      class="coming-soon"
    >
      <div
        class="logo"
      >
        <img
          alt="Logo"
          src="logo_dark.svg"
        />
      </div>
      <div
        class="countdown"
      >
        <div
          class="time"
        >
          296 
          <span>
            Days
          </span>
        </div>
        <div
          class="time"
        >
          15 
          <span>
            Hours
          </span>
        </div>
        <div
          class="time"
        >
          7 
          <span>
            Minutes
          </span>
        </div>
        <div
          class="time"
        >
          31 
          <span>
            Seconds
          </span>
        </div>
      </div>
      <div
        class="notify-me"
      >
        <p>
          We'll be coming soon!
        </p>
        <form>
          <input
            placeholder="Email address"
            type="email"
          />
          <button
            type="submit"
          >
            Notify Me
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;